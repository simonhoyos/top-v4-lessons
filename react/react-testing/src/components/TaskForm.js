import React from 'react';

class TaskForm extends React.Component {
  render() {
    const { onSubmit, onChange, title } = this.props
    return (
      <form
        data-testid="task-form"
        className="create-task"
        onSubmit={onSubmit}
      >
        <label htmlFor="title">TÃ­tulo</label>
        <input
          data-testid="input-title"
          id="title"
          name="title"
          onChange={onChange}
          type="text"
          value={title}
        />
        <button>Crear tarea</button>
      </form>
    )
  }
}

export default TaskForm;
