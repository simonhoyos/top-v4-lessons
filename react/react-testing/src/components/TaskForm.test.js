import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import TaskForm from './TaskForm';

const mockStore = configureStore();

describe('TaskForm', () => {
  it('should match snapshot', () => {
    const store = mockStore({})
    const wrapper = mount(
      <Provider store={store}>
        <TaskForm />
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  xit('should change input', () => {
    const onChange = jest.fn();
    const store = mockStore({
      onChange,
      onSubmit: jest.fn(),
      title: ''
    })
    const wrapper = mount(
      <Provider store={store}>
        <TaskForm />
      </Provider>
    );

    const input = wrapper.find('#title');
    input.simulate('change', {
      target: {
        name: 'title',
        value: 'lo que sea'
      }
    });

    expect(onChange).toHaveBeenCalled();
  });
});
