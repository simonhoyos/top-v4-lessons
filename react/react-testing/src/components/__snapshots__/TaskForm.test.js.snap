// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TaskForm)>
    <TaskForm
      onChange={[Function]}
      onSubmit={[Function]}
    >
      <form
        className="create-task"
        data-testid="task-form"
        onSubmit={[Function]}
      >
        <label
          htmlFor="title"
        >
          TÃ­tulo
        </label>
        <input
          data-testid="input-title"
          id="title"
          name="title"
          onChange={[Function]}
          type="text"
        />
        <button>
          Crear tarea
        </button>
      </form>
    </TaskForm>
  </Connect(TaskForm)>
</Provider>
`;
